yapf/yapflib/reformatter.py

```diff
@@ -394,6 +399,186 @@ def _AlignTrailingComments(final_lines):
       final_lines_index += 1
 
 
+def _AlignDictColon(final_lines):
```

Comment:
I know that this is similar to the `_AlignTrailingComments` function, but it's also much longer. Could you split some of the code out into separate functions to make it more readable?